### DISCUSSION TITLE
Hearth/Home Update No Connection

### DISCUSSION THREAD

**Discussion by lyk4ndna** [mbround18/valheim-docker#440] [General]
I figured I would put this out there if anyone else was having issues connecting to the container after the new Hearth/Home update. 

The server stopped and updated this morning and it was great. I went to log on at night though and it said version incompatible, so I had to update the server again. I did docker-compose down and then docker-compose up and it updated again. 

I went to connect again and it said connection failed. I could see that my computer tried to make an attempt to connect but for some reason something wasn't letting it through. I used nmap to check and see if my ports were still forwarding and they were. I ended up just deleting the entry on my router for port forwarding and re-adding it. 

After I did that everything connected correctly again.

**Comment by mbround18** [comment]
Hello @lyk4ndna I am currently testing the upgrade path on the hearth and home update,

---

**Update 1**: I had to fully delete everything in my Valheim directory and launch in Vanilla. ValheimPlus looks like its not quite ready for usage with the update.

**Reply by lyk4ndna** [reply]
I was running on vanilla, so didn’t have to do any deleting. I want to try out valheimplus though when it’s ready to go!

**Reply by spannerman79** [reply]
Actually you do.

The areas that you have explored - and in a certian radius around you - won't have new content generated.  Now if you have just started off newish or only explored a very small area then you should be good to go.

But if you have explored a fair chunk size then it would be best to wipe the world.  You can even create a fresh map with the same seed if you so desire to - you just cannot keep the map data that has been already created.

Its in full detail in https://youtu.be/BUxrOEVMoUQ - the _exact_ timestamp I can't remember.

**Reply by lyk4ndna** [reply]
We haven’t explored enough to merit a change, I think. We still haven’t gone into Plains and just got into some Mountain. 

If I do change to ValheimPlus from Vanilla, would that just be a docker compose change? Would I still keep my worlds and everything in it?

**Reply by mbround18** [reply]
@lyk4ndna yes :) we've made it simple to be able to mod your Valheim instance by changing the TYPE env variable from Vanilla to ValheimPlus.

**Reply by spannerman79** [reply]
@lyk4ndna - Hearth & Home introduces new materials used in cooking and that is also in mountain areas.  In one seed for a map my friends and I looked over the mnountain biome was connected to Meadows biome - "newbie" zone basically.

Also, the map generation is **500 metres** around you.  It isn't like playing minecraft where map generation is the servers set view distance.  If you are to take minecraft generation into account and compare it to Valheim its equivalent is approx is server view distance is set to **30** which is extremely unusual for even a vanilla minecraft server.

I do suggest though if V+ is not the _only_ mod you use that instead you set your `TYPE` env to `BepInEx` or `BepInExFull`. I've noticed more and more mods are starting to depend more on `BepInExFull` over `BepInEx`.

Yes it isn't a simple as switching to setting one env var.

Overall I suggest you talk with your friends to see if **only** V+ will be used or if other mods will be used.  Sometimes its better to "rip the band aid off fast" then slower and going the longer route in the end.

Edit:  Once such mod that is **very** handy with the H&H update is https://valheim.thunderstore.io/package/Advize/PlantEverything/

As you can see, once you show your friends a mod like that, knowing that instead of hunting bloody everywhere for berry bushes (example) once they collect enough berries they can plant the bush themselves and create and actual "farming" area.  Expecially coupled with the mod https://valheim.thunderstore.io/package/Nexus/FarmGrid/ (needs config tweaks to get working correctly - details are on the PlantEverything page) and then you can see what can be done for those that like to farm/plant.  Adds a bit of role play to the game too 😄

**Reply by lyk4ndna** [reply]
@mbround18 @spannerman79  Awesome! Thank you so much! Loving those two mods planteverything and farmgrid will save so much time. A little bit of Stardew in my Viking world. :)

**Reply by spannerman79** [reply]
No problem @lyk4ndna 

Also, if you have those that are WoW/MMO players like I do what I did to entice them over was to add https://valheim.thunderstore.io/package/RandyKnapp/EpicLoot/ - better images can be viewed over at https://www.nexusmods.com/valheim/mods/387

As Valheim is in early access I've personally found that you need to add mods in order to keep game playability going until Iron Gate does more work on Valheim.

As **always**, adding mods _can_ break your world save.  It doesn't always but it is better to be safe then sorry.  So ensure you remember this mantra; _[Backup, Backup, Backup](https://github.com/mbround18/valheim-docker/blob/main/README.md#auto-backup)!_

Also, ensure that you read the mod description just in case it has other dependencies too.

**Comment by spannerman79** [comment]
https://github.com/valheimPlus/ValheimPlus/issues/628#issuecomment-920988639

Until its out of alpha, you can check the `#alpha-releases` text channel in the V+ discord.

**Comment by spannerman79** [comment]
Approx 12 mins ago

https://github.com/valheimPlus/ValheimPlus/releases/tag/0.9.9
